src/khoj/routers/api_content.py

```diff
@@ -325,6 +330,65 @@ def get_content_types(request: Request, client: Optional[str] = None):
     return list(configured_content_types & all_content_types)
 
 
+@api_content.get("/all", response_model=Dict[str, str])
+@requires(["authenticated"])
+async def get_all_content(request: Request, client: Optional[str] = None, truncated: Optional[bool] = True):
```

Comment:
Add pagination to avoid heavy data streaming with larger knowledge bases.

Reply:
This seems to have improved rendering while testing with the 2k+ documents from the FRAMES eval knowledge base